src/hoc/withPopup.js

```diff
@@ -1,8 +1,53 @@
 import React from 'react';
 
-const withPopup = () => {
-    return () => <section>withPopup()</section>;
-}
+const withPopup = backgroundColor => {
+	return WrappedComponent => {
+		return class Popup extends React.Component {
+			state = {
+				isOpen: true,
+			};
+
+			setStyle() {
+				return {
+					position: "relative",
+					width: "200px",
+					border: "1px solid red",
+					margin: "0 auto",
+					padding: "10px ",
+					borderRadius: "3px",
+					backgroundColor: backgroundColor || "#ffffff",
+				};
+			}
+
+			setButton() {
```

Comment:
Raczej bym dodał `Style` tj. `setButtonStyle` - aby wskazać o co dokładnie chodzi. 
`setButton` - raczej mówi o ustawieniu przycisku, a nie o jego wyglądzie.

Zwróciłbym uwagę też na to, że to nie ustawia styklów dla buttona, a je zwraca dlatego lepszą nazwą będzie `getButtonStyle`. Trzeba patrzeć z perspektywy funkcji co robi i tak ją nazwać.